Fix limited range normalization

cosmin opened the pull request:
I noticed a discrepancy in the range normalization for 10-bit. The problem seems to be due to how the current code is first converting to 0-1.0 range and then normalizing in the 0-1 range.

The issue stems from the fact that 16/255 is not exactly equal to 64/1023. The end result is a small but detectable difference. I believe the easy solution is to normalize for limited range directly in the 0-1.0 conversion, by doing frame_data - limited_min / (limited_max - limited_min) which will ensure that 16 in 8bit and 64 in 10bit will both normalize to exactly 0.0 and 235 in 8bit and 940 in 10bit will normalize to precisely 1.0

I've stacked this with the previous pull request to get CI feedback on the right versions. I'll rebase it after #24 is landed.

Fixes #25

slhck commented:
Looks good to me! Should be rebased with changes from #24 and some other commits I made on `main`.

cosmin commented:
The tests didn't fail because of limited precision in the test. I bumped that up to 4 decimal places and regenerated the ground truth files. In the process I ran into a problem because the current script did not run on my Mac due to case insensitive filesystem and relying on case to separate AOM from the rest. So I updated the script to be explicit about the AOM sources. And also the regeneration for limited seemed to be missing so I added that as well.

slhck commented:
Sounds good! I gave you maintainer access via a team membership – please check if you can do the squash & merge yourself.
